llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2011,8 +2012,13 @@ Constant *SPIRVToLLVM::buildConstStoreRecursively(SPIRVType *const spvType, Type
     return ConstantArray::get(cast<ArrayType>(storeType), constElements);
   }
   // If the store was a bool or vector of bool, need to zext the storing value.
-  if (spvType->isTypeBool() || (spvType->isTypeVector() && spvType->getVectorComponentType()->isTypeBool()))
-    constStoreValue = ConstantExpr::getZExtOrBitCast(constStoreValue, storeType);
+  if (spvType->isTypeBool() || (spvType->isTypeVector() && spvType->getVectorComponentType()->isTypeBool())) {
+    if (constStoreValue->getType() != storeType) {
+      constStoreValue = ConstantFoldCastOperand(Instruction::ZExt, constStoreValue, storeType, m_m->getDataLayout());
+      if (!constStoreValue)
```

Comment:
I wonder if an assert might be more appropriate here? I think the constant fold should always succeed shouldn't it unless a non-const value is passed in (in which case asserting might be the most helpful thing to do since that would be unexpected).

Reply:
Changed to an assert.